src/serac/numerics/functional/tests/functional_qoi.cpp

```diff
@@ -356,6 +356,101 @@ void qoi_test(mfem::ParMesh& mesh, H1<p1> trial1, H1<p2> trial2, Dimension<dim>)
   check_gradient(f, U1, U2);
 }
 
+TEST(QoI, DependsOnVectorValuedInput)
+{
+  constexpr int p   = 2;
+  constexpr int dim = 3;
+
+  mfem::ParMesh& mesh = *mesh3D;
+
+  auto                        fec = mfem::H1_FECollection(p, dim);
+  mfem::ParFiniteElementSpace fespace(&mesh, &fec, dim);
+
+  mfem::ParGridFunction           U_gf(&fespace);
+  mfem::VectorFunctionCoefficient x_squared(dim, [](const mfem::Vector x, mfem::Vector& y) {
+    y    = 0.0;
+    y[0] = x[0] * x[0];
+  });
+  U_gf.ProjectCoefficient(x_squared);
+
+  mfem::HypreParVector* tmp = fespace.NewTrueDofVector();
+  mfem::HypreParVector  U   = *tmp;
+  U_gf.GetTrueDofs(U);
+
+  // Define the types for the test and trial spaces using the function arguments
+  using trial_space = H1<p, dim>;
+
+  Functional<double(trial_space)> f({&fespace});
+  f.AddVolumeIntegral(
+      DependsOn<0>{}, [&](auto /*x*/, auto u) { return norm(serac::get<0>(u)); }, mesh);
+
+  double exact_answer   = 512.0 / 3.0;  // \int_0^8 x^2 dx = 512/3
+  double relative_error = (f(U) - exact_answer) / exact_answer;
+  EXPECT_NEAR(0.0, relative_error, 1.0e-10);
+
+  delete tmp;
+}
+
+TEST(QoI, AddAreaIntegral)
+{
+  constexpr int p   = 1;
+  constexpr int dim = 2;
+
+  mfem::ParMesh& mesh = *mesh2D;
+
+  auto                        fec = mfem::H1_FECollection(p, dim);
+  mfem::ParFiniteElementSpace fespace(&mesh, &fec);
+
+  mfem::ParGridFunction     U_gf(&fespace);
+  mfem::FunctionCoefficient x_squared([](mfem::Vector x) { return x[0] * x[0]; });
+  U_gf.ProjectCoefficient(x_squared);
+
+  mfem::HypreParVector* tmp = fespace.NewTrueDofVector();
+  mfem::HypreParVector  U   = *tmp;
+  U_gf.GetTrueDofs(U);
+
+  // Define the types for the test and trial spaces using the function arguments
+  using trial_space = H1<p>;
+
+  Functional<double(trial_space)> measure({&fespace});
+  measure.AddAreaIntegral(
+      DependsOn<>{}, [&](auto /*x*/) { return 1.0; }, mesh);
+  double relative_error = (measure(U) - measure_mfem(mesh)) / measure(U);
```

Comment:
I may not have understood the intent of this test, so I apologize if this is off base: It looks like this test verifies that the Functional implementation gets the same answer as a native MFEM implementation. I think for unit tests we should prefer exact solutions to code-to-code comparisons. The former gives us a clearer signal if it fails.